packages/core/getBusyTimes.ts

```diff
@@ -144,43 +144,50 @@ export async function getBusyTimes(params: {
       });
 
   const bookingSeatCountMap: { [x: string]: number } = {};
-  const busyTimes = bookings.reduce(
-    (aggregate: EventBusyDetails[], { id, startTime, endTime, eventType, title, ...rest }) => {
-      if (rest._count?.seatsReferences) {
-        const bookedAt = `${dayjs(startTime).utc().format()}<>${dayjs(endTime).utc().format()}`;
-        bookingSeatCountMap[bookedAt] = bookingSeatCountMap[bookedAt] || 0;
-        bookingSeatCountMap[bookedAt]++;
-        // Seat references on the current event are non-blocking until the event is fully booked.
-        if (
-          // there are still seats available.
-          bookingSeatCountMap[bookedAt] < (eventType?.seatsPerTimeSlot || 1) &&
-          // and this is the seated event, other event types should be blocked.
-          eventTypeId === eventType?.id
-        ) {
-          // then we do not add the booking to the busyTimes.
-          return aggregate;
-        }
-        // if it does get blocked at this point; we remove the bookingSeatCountMap entry
-        // doing this allows using the map later to remove the ranges from calendar busy times.
-        delete bookingSeatCountMap[bookedAt];
-      }
-      if (rest.uid === rescheduleUid) {
-        return aggregate;
-      }
-      aggregate.push({
-        start: dayjs(startTime)
-          .subtract((eventType?.beforeEventBuffer || 0) + (afterEventBuffer || 0), "minute")
-          .toDate(),
-        end: dayjs(endTime)
-          .add((eventType?.afterEventBuffer || 0) + (beforeEventBuffer || 0), "minute")
-          .toDate(),
-        title,
-        source: `eventType-${eventType?.id}-booking-${id}`,
-      });
-      return aggregate;
-    },
-    []
-  );
+  // Only take bookings as source of truth when there are no connected calendars
+  // this is because cancellations downstream aren't reflected in our Booking database.
+  // TODO: Reflect downstream cancellations in our Booking database and use Booking as
+  //       source of truth.
+  const busyTimes =
+    credentials?.length > 0
```

Comment:
other non-calendar apps are also in credentials. So if there is no calendar but for example true we wouldn't get any busyTimes